README.md

```diff
@@ -0,0 +1,109 @@
+1. **tests were minimal and most testing was like "if it compiles, it's good"**
+   * The _OP_ suffered from only having a few tests, and even though Rust only compiles decent quality software, by nature, certain reasoning was just in my head. _'Why is it doing this particular thing? It seems wrong'_ would be impossible to know. 
+   * manual testing is slow and error prone
+1. **Versions like 1.2.0+2019-08-23...` would additionally show the version of the Google API description, but is ignored by `cargo`** 
+   * This was done to discriminate the 'code' version from the version of the API it represents.
+   * As the `+` is ignored by `cargo`, to re-release a crate with a new version of the API, one would have to increment the patch level. However, that would force all crates to be re-released, even if their API version didn't change at all.
+   * This caused unnecessary spamming of `crates.io`, and the `+` should be a `-` to fix this.
+1. **The 'fields' parameter could not be used to have smaller response types**
+   * Some [Response Types](https://docs.rs/google-sheets4/1.0.10+20190625/google_sheets4/struct.Response.html) are huge, even though with the right `field` setting, one would only receive a fraction of the data. However, one would always have to allocate big structures to with most of the optional fields set to `None`.
+   * 💡Idea 💡: Can [serde(flatten)](https://serde.rs/field-attrs.html#flatten) be used to subdivide the possible field sets in the data structure? Probably it's not known which actual fields belong to each `field` argument.
+
+
+# Technology and Architecture Sketches
+
+Items mentioned below ideally create a link to one of the problems they slove, e.g. `🥵2` , the project goal they support, e.g `🛸3`, or the effective thing they build on (`🌈1`).
+
+## Toolchains
+
+Here is the anticipated tooling. What follows is the list of tools I would add and why.
+
+* **make** - repeats `🌈1` 
+  * I am a fan of simple makefiles, which catch dependencies between files and run a script to generate them. This served _OP_ extremely well.
+  * get parallelization for free, and make transparent which programs to call and how to get work done.
+  * the Makefile serves as hub keeping all commands one would run to interact with the project in any way.
+* **Cargo/Rust** - fixes `🥵4`, support `🌈1`
+  * All work should be done by a Rust binary, which helps keeping things easy with `make`. Previously, the single, magical binary was `python`, and when adding even a single additional Rust tool, one would pay with some complexity. One Rust binary with multiple sub-command seems reasonable.
+* **rust-fmt** - helps remedy `🥵8`
+  * Definitely needed to get idiomatically looking code.
+  * _OP_ didn't have it, it wasn't a real problem, but too much time was spent making things look pretty. With `rust-fmt`, templates can be optimized for maintainability, even if the output doesn't look great initially.
```

Comment:
yes, I would encourage generating the code in the simplest way possible and living with whatever output rustfmt chooses. Users understand this is generated code when looking at the source, and no amount of hand curating will ever change that.